oximeter/instruments/src/http.rs

```diff
@@ -93,6 +95,56 @@ impl RequestLatencyHistogram {
     }
 }
 
+/// A helper trait to extract a status code from a Dropshot response.
+pub trait TrackableResponse: HttpResponse {
+    fn status_code(&self) -> StatusCode;
```

Comment:
What if we add this to dropshot::HttpResponse?

Reply:
That would be nice. The previous code relied on the static method `HttpResponse::response_metadata()`, which doesn't work for a `Response<Body>`, where you need `self` to return the actual status code. Adding that to the existing trait would be neater solution than this one.

Reply:
I've taken this approach in https://github.com/oxidecomputer/dropshot/pull/1049, so I'll pull that update in here, and shrink this PR accordingly.

Reply:
Updated in 0dee1893d